src/mapgen/mapgen_v6.cpp

```diff
@@ -865,8 +857,6 @@ void MapgenV6::flowMud(s16 &mudflow_minpos, s16 &mudflow_maxpos)
 						continue;
 					// Check that under side is air
 					VoxelArea::add_y(em, i2, -1);
-					if (!vm->m_area.contains(i2))
-						continue;
```

Comment:
Unnecessary check removed because:
The previous 'out of voxelmanip' check a few lines above ensures this column is horizontally inside the voxelmanip.
y is somewhere between node_min.Y and node_max.Y, so y - 1 is well within the voxelmanip.

Reply:
No it doesn’t. `max_spread_amount` does (unless I missed something), it is one less than the mapblock size specifically for that, to let you step that one node away fearlessly. Otherwise wrong node would be accessed sometimes.

Remember: `VoxelArea::contains` only checks that some linear combination λ = αx + βy + γz + δ is a valid index. If (x, y, z) is inside the area, λ is actually index of the node at that position. But that position is non-unique; one step out of area usually gives you such an “alternative” position, for which λ is the same as for some node in area.

Reply:
Good point, thanks.
Due to the value of `max_spread_amount` the `// Move to side` check above isn't really needed, but i left it in to be safe, maybe i should remove it too. Will think on this further, however the PR is still good as it is.

Reply:
These 'area contains index' checks aren't very reliable then. Will reconsider, WIP.

Reply:
All attended to now.